atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/sweep/metrics/SweepMetricsManager.java

```diff
@@ -15,42 +15,42 @@
 import com.palantir.atlasdb.AtlasDbMetricNames;
 import com.palantir.atlasdb.keyvalue.api.SweepResults;
-import com.palantir.atlasdb.keyvalue.api.TableReference;
-import com.palantir.atlasdb.logging.LoggingArgs;
+import com.palantir.atlasdb.util.MetricsManager;
 
 // Not final for Mockito
 @SuppressWarnings("checkstyle:FinalClass")
 public class SweepMetricsManager {
-    private final SweepMetricsFactory factory = new SweepMetricsFactory();
-
-    private final SweepMetric<Long> cellsExamined = factory.accumulatingLong(AtlasDbMetricNames.CELLS_EXAMINED);
-    private final SweepMetric<Long> cellsDeleted = factory.accumulatingLong(AtlasDbMetricNames.CELLS_SWEPT);
-    private final SweepMetric<Long> timeSweeping = factory.simpleLong(AtlasDbMetricNames.TIME_SPENT_SWEEPING);
-    private final SweepMetric<Long> totalTime = factory.simpleLong(AtlasDbMetricNames.TIME_ELAPSED_SWEEPING);
-    private final SweepMetric<String> tableSweeping = factory.simpleString(AtlasDbMetricNames.TABLE_BEING_SWEPT);
-    private final SweepMetric<Long> sweepErrors = factory.simpleMeter(AtlasDbMetricNames.SWEEP_ERROR);
-
-    public void resetBeforeDeleteBatch() {
-        cellsExamined.set(0L);
-        cellsDeleted.set(0L);
-    }
+    public static final String METRIC_BASE_NAME = SweepMetricsManager.class.getPackage().getName() + ".SweepMetric";
+
+    private final MetricRegistry metricRegistry = new MetricsManager().getRegistry();
+
+    private final Counter cellsExamined = metricRegistry.counter(getMetricName(AtlasDbMetricNames.CELLS_EXAMINED));
+    private final Counter cellsDeleted = metricRegistry.counter(getMetricName(AtlasDbMetricNames.CELLS_SWEPT));
+    private final Counter timeSweeping = metricRegistry.counter(getMetricName(AtlasDbMetricNames.TIME_SPENT_SWEEPING));
+    private final Counter totalTime = metricRegistry.counter(getMetricName(AtlasDbMetricNames.TIME_ELAPSED_SWEEPING));
+    private final Counter sweepErrors = metricRegistry.counter(getMetricName(AtlasDbMetricNames.SWEEP_ERROR));
 
-    public void updateAfterDeleteBatch(long cellTsPairsExamined, long staleValuesDeleted) {
-        cellsExamined.accumulate(cellTsPairsExamined);
-        cellsDeleted.accumulate(staleValuesDeleted);
+
+    public void updateCellExaminedDeleted(long cellTsPairsExamined, long staleValuesDeleted) {
+        cellsExamined.inc(cellTsPairsExamined);
+        cellsDeleted.inc(staleValuesDeleted);
     }
 
-    public void updateMetrics(SweepResults sweepResults, TableReference tableRef) {
-        cellsExamined.set(sweepResults.getCellTsPairsExamined());
-        cellsDeleted.set(sweepResults.getStaleValuesDeleted());
-        tableSweeping.set(LoggingArgs.safeTableOrPlaceholder(tableRef).getQualifiedName());
-        timeSweeping.set(sweepResults.getTimeInMillis());
-        totalTime.set(sweepResults.getTimeElapsedSinceStartedSweeping());
+    public void updateSweepTime(long sweepTime, long totalTimeElapsedSweeping) {
```

Comment:
`totalTime` is already an absolute value at the sweep result level, so you should use `CurrentValueMetric<Long>` instead of the Counter for it and just set to the appropriate value. Accumulating would yield nonsensical results